daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v1/Runner.scala

```diff
@@ -0,0 +1,200 @@
+                            )
+                          ) =>
+                        val e =
+                          SELet1(
+                            SEImportValue(typ, value),
+                            SELet1(
+                              SEAppAtomic(SEBuiltin(SBToAny(typ)), Array(SELocS(1))),
+                              SEAppAtomic(SEValue(handle), Array(SELocS(1))),
+                            ),
+                          )
+                        machine.setExpressionToEvaluate(e)
+                        stepToValue()
+                          .fold(Future.failed, Future.successful)
+                          .flatMap {
+                            case SOptional(None) =>
+                              Future.failed(exce)
+                            case SOptional(Some(free)) =>
+                              Future.successful(SEApp(SEValue(continue), Array(free)))
+                            case e =>
+                              Future.failed(
+                                new ConverterException(s"Expected SOptional but got $e")
+                              )
+                          }
+                      case Failure(e) => Future.failed(e)
+                    }
+                  case ScriptF.Throw(SAny(ty, value)) =>
+                    Future.failed(
+                      Runner.InterpretationError(
+                        SError
+                          .SErrorDamlException(IE.UnhandledException(ty, value.toUnnormalizedValue))
+                      )
+                    )
+                  case cmd: ScriptF.Cmd =>
+                    cmd.execute(env).transform {
+                      case f @ Failure(Runner.CanceledByRequest) => f
+                      case f @ Failure(Runner.TimedOut) => f
+                      case Failure(exception) =>
+                        Failure(new Script.FailedCmd(cmd, exception))
+                      case Success(value) =>
+                        Success(value)
+                    }
+                }
+              }
+              .flatMap(run(_))
+          case Left(v) =>
+            v match {
+              case SRecord(_, _, ArrayList(newState, _)) => {
+                // Unwrap the Tuple2 we get from the inlined StateT.
+                Future { newState }
+              }
+              case _ => Future.failed(new ConverterException(s"Expected Tuple2 but got $v"))
+            }
+        }
+    }
+
+    val ideLedgerContext: Option[IdeLedgerContext] =
+      initialClientsV1.default_participant match {
+        case Some(ledgerClient: ledgerinteraction.IdeLedgerClient) =>
+          Some(new IdeLedgerContext {
+            override def currentSubmission: Option[ScenarioRunner.CurrentSubmission] =
+              ledgerClient.currentSubmission
+            override def ledger: ScenarioLedger = ledgerClient.ledger
+          })
+        case _ => None
+      }
```

Comment:
We now package up the ide ledgers submission and ledger fields into this trait which is unversioned (but still "lazy")